vault/resource_jwt_auth_backend.go

```diff
@@ -192,37 +233,9 @@ func jwtAuthBackendUpdate(d *schema.ResourceData, meta interface{}) error {
 	path := getJwtPath(d)
 	log.Printf("[DEBUG] Updating auth %s in Vault", path)
 
-	configuration := map[string]interface{}{
-		"oidc_discovery_ca_pem": d.Get("oidc_discovery_ca_pem"),
-		"bound_issuer":          d.Get("bound_issuer"),
-	}
-
-	oidcDiscoveryUrl, oidcDiscoveryUrlExists := d.GetOk("oidc_discovery_url")
-	jwtValidationPubKeys, jwtValidationPubKeysExists := d.GetOk("jwt_validation_pubkeys")
-
-	if oidcDiscoveryUrlExists == jwtValidationPubKeysExists {
-		return errors.New("exactly one of oidc_discovery_url and jwt_validation_pubkeys should be provided")
-	}
-
-	if oidcDiscoveryUrlExists {
-		configuration["oidc_discovery_url"] = oidcDiscoveryUrl
-	}
-
-	if jwtValidationPubKeysExists {
-		configuration["jwt_validation_pubkeys"] = jwtValidationPubKeys
-	}
-
-	if v, ok := d.GetOkExists("jwt_supported_algs"); ok {
-		configuration["jwt_supported_algs"] = v
-	}
-	if v, ok := d.GetOkExists("oidc_client_id"); ok {
-		configuration["oidc_client_id"] = v.(string)
-	}
-	if v, ok := d.GetOkExists("oidc_client_secret"); ok {
-		configuration["oidc_client_secret"] = v.(string)
-	}
-	if v, ok := d.GetOkExists("default_role"); ok {
-		configuration["default_role"] = v.(string)
+	configuration := map[string]interface{}{}
+	for _, configOption := range matchingJwtMountConfigOptions {
+		configuration[configOption] = d.Get(configOption)
```

Comment:
This logic appears subtly different from the logic above. Above, most of the parameters are only set _if they exist_ whereas here they're just set. I'm concerned that this might cause the update to behave differently than before.

Reply:
Good point, I'll adjust the logic here.